openedx/core/lib/api/tests/test_authentication.py

```diff
@@ -164,72 +205,85 @@ def test_post_form_passing_auth_url_transport(self):
-        """Ensure POSTing with a readonly scope instead of a write scope fails"""
-        read_only_access_token = self.access_token
-        read_only_access_token.scope = oauth2_provider_scope.SCOPE_NAME_DICT['read']
-        read_only_access_token.save()
-        auth = self._create_authorization_header(token=read_only_access_token.token)
-        response = self.csrf_client.get('/oauth2-with-scope-test/', HTTP_AUTHORIZATION=auth)
-        self.assertEqual(response.status_code, 200)
-        response = self.csrf_client.post('/oauth2-with-scope-test/', HTTP_AUTHORIZATION=auth)
-        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+        response = self.post_with_bearer_token('/oauth2-test/')
+        self.check_error_codes(
+            response,
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            error_code=authentication.OAUTH2_TOKEN_ERROR_EXPIRED
+        )
 
+    @ddt.data(
+        TokenError('notatoken', authentication.OAUTH2_TOKEN_ERROR_NONEXISTENT),
+        TokenError('malformed token', authentication.OAUTH2_TOKEN_ERROR_MALFORMED),
+        TokenError('', authentication.OAUTH2_TOKEN_ERROR_NOT_PROVIDED),
+    )
+    def test_response_for_post_request_with_bad_auth_token(self, token_error):
+        response = self.post_with_bearer_token('/oauth2-test/', token=token_error.token)
+        self.check_error_codes(response, status_code=status.HTTP_401_UNAUTHORIZED, error_code=token_error.error_code)
+
+    @ddt.data(
+        ScopeStatuses('read', read_status=status.HTTP_200_OK, write_status=status.HTTP_403_FORBIDDEN),
+        ScopeStatuses('write', status.HTTP_403_FORBIDDEN, status.HTTP_200_OK),
+    )
     @unittest.skipUnless(oauth2_provider, 'django-oauth2-provider not installed')
-    def test_post_form_with_valid_scope_passing_auth(self):
-        """Ensure POSTing with a write scope succeed"""
-        read_write_access_token = self.access_token
-        read_write_access_token.scope = oauth2_provider_scope.SCOPE_NAME_DICT['write']
-        read_write_access_token.save()
-        auth = self._create_authorization_header(token=read_write_access_token.token)
-        response = self.csrf_client.post('/oauth2-with-scope-test/', HTTP_AUTHORIZATION=auth)
-        self.assertEqual(response.status_code, 200)
+    def test_responses_to_scoped_requests(self, scope_statuses):
+        previous_scope = self.access_token.scope
+        self.access_token.scope = oauth2_provider_scope.SCOPE_NAME_DICT[scope_statuses.scope]
+        self.access_token.save()
+        response = self.get_with_bearer_token('/oauth2-with-scope-test/', token=self.access_token.token)
+        self.assertEqual(response.status_code, scope_statuses.read_status)
+        response = self.post_with_bearer_token('/oauth2-with-scope-test/', token=self.access_token.token)
+        self.assertEqual(response.status_code, scope_statuses.write_status)
+        self.access_token.scope = previous_scope
+        self.access_token.save()
```

Comment:
Oops.  They aren't.  I was seeing behavior that I wasn't expecting, and thought that would fix it, but it turned out the problem was in my expectations.  I'll remove those (As well as L281)